cpp/daal/src/algorithms/linear_model/linear_model_train_normeq_update_impl.i

```diff
@@ -173,6 +174,55 @@ Status UpdateKernel<algorithmFPType, cpu>::compute(const NumericTable & xTable,
     return UpdateKernel<algorithmFPType, cpu>::compute(xTable, yTable, xtxTable, xtyTable, initializeResult, interceptFlag, nullptr);
 }
 
+template <typename algorithmFPType, CpuType cpu>
+Status computeNonBatchedAggregates(const DAAL_INT nRows, const DAAL_INT nCols, const DAAL_INT nResponses, bool initializeResult, bool interceptFlag,
+                                   const algorithmFPType * xPtr, const algorithmFPType * yPtr, algorithmFPType * xtx, algorithmFPType * xty)
+{
+    DAAL_INT nBetasIntercept = nCols + static_cast<int>(interceptFlag);
+    DAAL_INT one_int         = 1;
+    algorithmFPType one      = 1;
+    algorithmFPType zero     = 0;
+    std::unique_ptr<algorithmFPType[]> ones;
+    if (interceptFlag)
+    {
+        ones = std::unique_ptr<algorithmFPType[]>(new algorithmFPType[nRows]);
+        std::fill(ones.get(), ones.get() + nRows, algorithmFPType(1));
+    }
+
+    BlasInst<algorithmFPType, cpu>::xsyrk("U", "N", &nCols, &nRows, &one, xPtr, &nCols, &zero, xtx, &nBetasIntercept);
+    if (interceptFlag)
+    {
+        BlasInst<algorithmFPType, cpu>::xgemv("N", &nCols, &nRows, &one, xPtr, &nCols, ones.get(), &one_int, initializeResult ? &zero : &one,
```

Comment:
Please move this `initializeResult ? &zero : &one` into some pre-computed value.
This code duplication makes it harder to read.